### ISSUE TITLE
Exploding gradient problem

### ISSUE THREAD

**Issue by scarmaten** [qfgaohao/pytorch-ssd#47]
I have tried training on open image dataset but i kept getting exploding gradient after 1 epoch
![photo_2019-04-22_01-54-36](https://user-images.githubusercontent.com/12598612/56473721-b6268200-64a1-11e9-9e86-157a149790e1.jpg)

**Comment by qfgaohao** [comment]
@scarmaten have you tried smaller learning rates?

**Comment by scarmaten** [comment]
yup i have tried it. still no luck

**Comment by qfgaohao** [comment]
Hi @scarmaten , one possible way to debug is use a very small, verified dataset as both train and val data to train overfitted models. If you can get a overfitted model with 100% (or almost) accuracy, that means the architecture and system is fine, and the problem might be the data. Otherwise, there is something wrong in the system design.

**Comment by tamyiuchau** [comment]
I have also come across exploding gradient problem when trying to use mobilenet v2 ssd on Widerface. I am able to bisect the problem to the smooth L1 loss at https://github.com/qfgaohao/pytorch-ssd/blob/1e86eb3a9655ca1893b537829bbbc2794614d485/vision/nn/multibox_loss.py#L45  .
I suspect it is caused by no default boxes matched. After changing torch.nn.functional.smooth_l1_loss( reduction='_**sum**_'), problem is solved.

**Comment by tamyiuchau** [comment]
The question then arise, should I adjust the prior boxes for my specific purposes, or just let it go?(As it is kind of working now)
